### PR TITLE
improve tools/souffle_converter.py 

### PR DESCRIPTION
**Pull request by Price1999a**
Improved `tools/souffle_converter.py `in three aspects
1. Fixed the file path problem: Doop now puts facts files under `/database`
2. Fixed the conversion of `ord() `function: the original script did not perform type conversion
3. Fixed the problem that the output declaration in the component cannot be processed when the parameter `--facts-only` is used

For the third fix, the overall idea is to collect all output declarations in `Relation.dumpOrder`.

[2-object-sensitive+heap.dl.zip](https://github.com/vmware/differential-datalog/files/7544084/2-object-sensitive%2Bheap.dl.zip)

I used the above souffle program for testing. This program comes from the corresponding analysis rules in Doop and made equivalent modifications. The original version is attached below:

[2-object-sensitive+heap-raw.dl.zip](https://github.com/vmware/differential-datalog/files/7544105/2-object-sensitive%2Bheap-raw.dl.zip)

### CODE REVIEW

**Comment by mbudiu-vmw** [comment]
This is great. To test the souffle converter we run the program in `./test` called `./run-souffle-tests-in-batches.py`
The souffle_release that is tested is hardwired in the script called by that one called `run-souffle-tests.py`.
Can you make sure that this test runs correctly?
The CI infrastructure will attempt to run it anyway.

**Comment by mbudiu-vmw** [comment]
If you have trouble running that script please let me know.
You will notice that there are lots of exceptions hardwired in the program run-souffle-tests.py, these are all souffle standard tests which fail to compile or execute correctly. If we move up the revision tested we may need to update this list too...

**Comment by Price1999a** [comment]
I am running `./run-souffle-tests-in-batches.py` 
CI have passed
https://github.com/Price1999a/differential-datalog/actions

**Comment by mbudiu-vmw** [comment]
Unfortunately the tests are relatively slow due to the very slow rust compiler. Let me know if the batches tests have passed too. We are actually not running all CI tests on github, because they take too long, and I believe that these souffle tests are only run on some paid gitlab machines, under some specific circumstances. If they pass for you it should be easy to merge them.
If they don't I may have some work to do to figure out why.

**Comment by Price1999a** [comment]
My server seems to have some network connection problems, I will try to run it again later.

```
Running from 0
Starting at 07:43:18.661169
Testing example/ackermann
Output relation Rack
Output relation Rack
Testing example/andersen
Output relation RPointsTo
Output relation RPointsTo
Testing example/bigrams
Output relation Rbigram
Output relation Rbigram
Testing example/cba_expr_value
Output relation Rdata_term
Output relation Rdata_term
svn: E170013: Unable to connect to a repository at URL 'https://github.com/souffle-lang/souffle/tags/2.0.2/tests/example/centroids'
svn: E120107: Error running context: The proxy server returned an error while setting up the SSL tunnel.
Testing example/clique
Output relation Rreachable
Output relation Rsame_clique
Output relation Redge
Output relation Rreachable
Output relation Rsame_clique
Output relation Redge
Converted successfully 5 out of 5 skipped xfail 9 running 5 after skipping 0
Compiling Rust at 08:08:17.658854
    Updating crates.io index
warning: spurious network error (2 tries remaining): [28] Timeout was reached (Proxy CONNECT aborted due to timeout); class=Net (12)
warning: spurious network error (1 tries remaining): [28] Timeout was reached (Proxy CONNECT aborted due to timeout); class=Net (12)
error: failed to get `libc` as a dependency of package `cmd_parser v0.1.0 (/home/njuseg/ddlog/test/souffle_tests_ddlog/cmd_parser)`

Caused by:
  failed to load source for dependency `libc`

Caused by:
  Unable to update registry `crates-io`

Caused by:
  failed to fetch `https://github.com/rust-lang/crates.io-index`

Caused by:
  network failure seems to have happened
  if a proxy or similar is necessary `net.git-fetch-with-cli` may help here
  https://doc.rust-lang.org/cargo/reference/config.html#netgit-fetch-with-cli

Caused by:
  [28] Timeout was reached (Proxy CONNECT aborted due to timeout); class=Net (12)
Terminating with exit code 1
Terminating with exit code 1
```

**Comment by mbudiu-vmw** [comment]
I will be running these tests as well.
The script was designed to be restartable to work around such problems.
It runs tests in batches of 5, and if you supply a number it will start from that point only.

File: tools/souffle_converter.py [hunk 1]
```diff
@@ -1663,6 +1666,15 @@ def process(self, tree: parglare.NodeNonTerm) -> None:
                 self.process_decl(decl)
         self.files.output(self.aggregates)
 
+    def preprocess(self, tree: parglare.NodeNonTerm) -> None:
+        decls = getListField(tree, "Declaration", "DeclarationList")
+        for decl in decls:
+            # if self.conversion_options.skip_logic:
```
Comment (mbudiu-vmw) [inline tools/souffle_converter.py:1672]: this should probably be deleted if it is not needed.

**Review by mbudiu-vmw — APPROVED** [review]

**Comment by mbudiu-vmw** [comment]
I am currently running the tests on release 2.0.2 of Souffle.
If they pass I will merge this.
However, we may want to try on newer releases too, but we may encounter other problems.

**Pull request merged**